File: 시저암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    string low = "abcdefghijklmnopqrstuvwxyz";
    string up;
    for(auto ch : low){
        up+=toupper(ch);
    }
    int baseLen = low.size();
    for(auto ch : s){
        if (ch == ' ')
        {
            /* code */
            answer+=ch;
        }else if(islower(ch)){
            int idx = low.find(ch);
            idx+=n;
            answer+=low[idx%baseLen];
        }else if(isupper(ch)){
            int idx = up.find(ch);
            idx+=n;
            answer+=up[idx%baseLen];
        }
    }
    return answer;
}

File: 시저암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";
		String low = "abcdefghijklmnopqrstuvwxyz";
		String up = low.toUpperCase();
		int baseLen = low.length();
		
		char[] lower = low.toCharArray();
		char[] upper = up.toCharArray();
		for(int i=0,len=s.length();i<len;i++){
			if (s.charAt(i)==' ')
				answer+=' ';
			else if (low.indexOf(s.charAt(i))!=-1){
				int idx = low.indexOf(s.charAt(i));
				idx+=n;
				answer+=lower[idx%baseLen];
			}else if(up.indexOf(s.charAt(i))!=-1){
				int idx = up.indexOf(s.charAt(i));
				idx+=n;
				answer+=upper[idx%baseLen];
			}
		}
		return answer;
    }
}
